src/models/replicate.js

```diff
@@ -0,0 +1,64 @@
+import Replicate from 'replicate';
+import { toSinglePrompt } from '../utils/text.js';
+
+// llama, mistral
+export class ReplicateAPI {
+	constructor(model_name, url) {
+		this.model_name = model_name;
+		this.url = url;
+
+		if (this.url) {
+			console.warn('Replicate API does not support custom URLs. Ignoring provided URL.');
+		}
+
+		if (!process.env.REPLICATE_API_KEY) {
+			throw new Error('Replicate API key missing! Make sure you set your REPLICATE_API_KEY environment variable.');
+		}
+
+		this.replicate = new Replicate({
+			auth: process.env.REPLICATE_API_KEY,
+		});
+	}
+
+	async sendRequest(turns, systemMessage) {
+		const stop_seq = '***';
+		let prompt_template;
+		const prompt = toSinglePrompt(turns, systemMessage, stop_seq);
+		let model_name = this.model_name || 'meta/meta-llama-3-70b-instruct';
+		if (model_name.includes('llama')) { // llama
```

Comment:
Not all llama models use this prompt template

Reply:
dude I don't understand prompt templates. I can't find docs for them, and it seems like a real headache to keep them specific to whatever model you're using. I just copied the ones over from docs from the models that work best. if you have a better idea lmk